sam-react-app/coreconcept/src/counter/Counter.tsx

```diff
@@ -0,0 +1,75 @@
+import React, {Component, createElement} from "react";
+
+class Counter extends Component {
+    state = {
+        counter: 0,
+    };
+    handleOnClick = (event: Event) => {
+        const element = event?.target as HTMLElement;
+        const componentId = element.getAttribute('id');
```

Comment:
Normally you don't want to rely on elements attributes. This should be resolved with JS: you can create multiple handlers `increment` and `decrement` or add a param `handleOnClick = (diff: number)` and you can call it ` {onChange: () => this.handleInputChange(1)},` for increment and with `-1` for decrement.